core/remotes/resolver.go

```diff
@@ -49,6 +49,10 @@ type Resolver interface {
 	// The returned Pusher should satisfy content.Ingester and concurrent attempts
 	// to push the same blob using the Ingester API should result in ErrUnavailable.
 	Pusher(ctx context.Context, ref string) (Pusher, error)
+
+	// Lister returns a new lister for the provided reference.
+	// The returned Lister can list tags for a given remote + hosts + reference.
+	Lister(ctx context.Context, ref string) (Lister, error)
```

Comment:
I'm a bit on the fence if this is something that should be part of the `Resolver` interface; adding to this interface can be a breaking change, but also a bit on the fence if discovery should be part of this interface altogether 🤔 


I had a quick  peek in some code-bases, and found at least 4 different implementations that currently  satisfy this interface; expanding the interface means that those would no longer match. I have not looked in depth where the interface is checked, but it's something that would have to be looked at.

Reply:
Hmmm. Good thought. It could be something separate but I wonder how it could be hooked into the general system then. Wdyt?